### DISCUSSION TITLE
Getting `[FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:43163/z9bjyARTgZU=/ws' when trying to run flutter test using appium-flutter-driver

### DISCUSSION THREAD

**Discussion by jagadeeshpurplle** [appium/appium#17508] [Q&A]
Hi Team, 

I'm trying to automate the flutter app using **appium-flutter-driver**. 

First & foremost Here is the source code of my test folder: https://drive.google.com/file/d/1jTiNYMLQIKQYGj6WBNrClpqpBnf2trv_/view?usp=sharing
Source code structure looks like this:
<img width="282" alt="Screenshot 2022-09-20 at 6 26 29 PM" src="https://user-images.githubusercontent.com/25878053/191263564-f169744f-4f22-4fbe-88b0-b6f599392cd0.png">

**App source is code nothing but, the sample flutter project provided by google it self.**

**Using below versions**
Appium - 1.22.3
npm version - 8.15.0
OS - Mac Monterey 12.3

**My first question is** 
- is there any certain things need to be added in development code in order to work for automation with appium-flutter-driver? like `enableFlutterDriverExtension()`,etc?

**Actual issue:** 
- When i try to execute the tests using **`APPIUM_OS=android npm start`**, i'm getting `**unknown error: An unknown server-side error occurred while processing the command. Original error: Failed to connect 30 times. Aborting.**`

**Here are the appium-cli logs:** https://gist.github.com/jagadeeshpurplle/456a20ff2d462ee23342e44b77a861b9

Can anybody helps me here to fix the issue please?

NOTE: When i try to run the same application with the app `flutterlogin-debug.apk` from https://github.com/gihanchandana46/AppiumFlutterDriver, then its working fine.
 

@KazuCocoa

**Comment by KazuCocoa** [comment]
Did you try out the latest appium-flutter-driver with appium2? I have added a couple of improvements to it. Potentially some of them help.

**Reply by IOnilearan** [reply]
I have the same issue @KazuCocoa I have the latest on both, do you know what I am missing?

**Reply by KazuCocoa** [reply]
i guess the local port forward did not work, or the server refused the connection.
What happen by attaching your local websocket client to connect to `ws://127.0.0.1:43163/z9bjyARTgZU=/ws` (just an example) which the appium log shows while appium flutter driver trying to attach the URL?
e.g. `wscat` in NodeJS

**Reply by IOnilearan** [reply]
hello @KazuCocoa it says this:

2022-09-30 11:29:02:798 - [FlutterDriver] Attempt #1
2022-09-30 11:29:02:799 - [FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:8280/ws
2022-09-30 11:29:02:809 - [FlutterDriver] [object Object]
2022-09-30 11:29:02:809 - [FlutterDriver] ex.target = [object Object]
2022-09-30 11:29:02:809 - [FlutterDriver] ex.type = error
2022-09-30 11:29:02:809 - [FlutterDriver] ex.message = read ECONNRESET
2022-09-30 11:29:02:810 - [FlutterDriver] ex.error = Error: read ECONNRESET
2022-09-30 11:29:02:810 - [FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:8280/ws
2022-09-30 11:29:02:810 - [FlutterDriver] Waiting 20 seconds before trying...
2022-09-30 11:29:22:819 - [FlutterDriver] Attempt #2
2022-09-30 11:29:22:819 - [FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:8280/ws
2022-09-30 11:29:22:824 - [FlutterDriver] [object Object]
2022-09-30 11:29:22:824 - [FlutterDriver] ex.target = [object Object]
2022-09-30 11:29:22:824 - [FlutterDriver] ex.type = error
2022-09-30 11:29:22:824 - [FlutterDriver] ex.message = read ECONNRESET
2022-09-30 11:29:22:824 - [FlutterDriver] ex.error = Error: read ECONNRESET

**Reply by IOnilearan** [reply]
It must be the port forwarding i think. do you know any examples of how i can implement this? @KazuCocoa

**Reply by jagadeeshpurplle** [reply]
@IOnilearan  does your app have enableFlutterDriverExtension() enabled in main.dart file before runApp()?

**Reply by jagadeeshpurplle** [reply]
For me i could able resolve this issue by updating the appium to `2.0.0-beta.44` and my desired capabilities to
`        'appium:platformName': '<PlatformName>',
        'appium:deviceName': '<DeviceName>',
        'appium:app': '<path>',
        'appium:autoGrantPermissions': 'true',
        'appium:automationName': 'Flutter'`

**Reply by IOnilearan** [reply]
@jagadeeshpurplle yes I do have this enabled, did you have to do any port forwarding?

**Reply by jagadeeshpurplle** [reply]
@IOnilearan No, i didn't do any port forwarding. I just started my appium with command ` appium --base-path /wd/hub `

**Reply by KazuCocoa** [reply]
I guess you created a session with something specific capabilities...?
https://github.com/appium-userland/appium-flutter-driver/blob/main/driver/lib/sessions/android.ts#L32-L44
I wonder if you set `observatoryWsUri`. Could you create a ticket in the repository with the entire appium server log?

**Reply by IOnilearan** [reply]
Yes so im trying to achieve the organised .specs of wdio so my capabilities are in a config.capabilities so the can run spec files. It's possible that may be the case due to this. I'll make a ticket

**Reply by IOnilearan** [reply]
@KazuCocoa here is the issue https://github.com/appium/appium/issues/17556
